### ISSUE TITLE
Can't decompile non-latest snapshots

### ISSUE THREAD

**Issue by Jiingy** [ShaneBeeStudios/McDeob#41]
I could decompile the latest snapshot fine (currently 24w05a), but not 24w04a. 

Command used:
java -jar McDeob-2.3.2.jar -version 24w04a -type client -decompile

**Comment by ShaneBeee** [comment]
these are the supported version:
found from the `-versions` command
```
 - Latest_Release
 - Latest_Snapshot
 - 1.20.4
 - 1.20.3
 - 1.20.2
 - 1.20.1
 - 1.20
 - 1.19.4
 - 1.19.3
 - 1.19.2
 - 1.19.1
 - 1.19
 - 1.18.2
 - 1.18.1
 - 1.18
 - 1.17.1
 - 1.17
 - 1.16.5
 - 1.16.4
 - 1.16.3
 - 1.16.2
 - 1.16.1
 - 1.16
 - 1.15.2
 - 1.15.1
 - 1.15
 - 1.14.4
 - Combat_Test_8c
 - 23w13TheVoteUpdate
 - 22w13oneBlockAtATime
 - 20w14infinite
 ```

**Comment by Jiingy** [comment]
Is it possible to add support for snapshots other than the latest?

**Comment by ShaneBeee** [comment]
at this time no.
I had done it before, manually, and it pushed me to have to update this app every week.
Someone had done a PR that stripped that out, and added in the "latest" versions, which fetches automatically from Mojang's servers.

I personally don't know why someone would want to decompile last weeks snapshot.
In the future I may revamp this and have it pull all versions from the server, but this won't be happening anytime soon.

**Comment by Jiingy** [comment]
For me, my personal intention with previous snapshots was to be able to compare changes to previous versions that I missed.

**Comment by ShaneBeee** [comment]
I did a major re-write to the way this program works.
Rather than me having to hardcode in each and every version, as well as it only pulling automatically the latest release/snapshot, this will pull ALL available releases/snapsots.

Check out the PR https://github.com/ShaneBeeStudios/McDeob/pull/43 if you want to test the build (found in the actions)

**Comment by ShaneBeee** [comment]
closing as this is now "fixed" in 3.0.0

**Comment by Jiingy** [comment]
Thanks!
